Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -463,6 +463,35 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     if shouldSkip(node) {
       return .skipChildren
     }
+    if let unexpected = node.unexpectedBetweenConditionAndTrailingComma,
+      let availability = node.condition.as(AvailabilityConditionSyntax.self),
+      let (comparisonOperator, falseKeyword) = unexpected.twoTekens(
+        firstSatisfying: { $0.tokenKind == .binaryOperator("==") },
+        secondSatisfying: { $0.tokenKind == .keyword(.false) }
+      )
+    {
+      // Diagnose #available used as an expression
+      let availabilityKeyword = availability.availabilityKeyword
+      let negatedAvailabilityKeyword = availabilityKeyword.negatedAvailabilityKeyword
+      addDiagnostic(
+        unexpected,
+        AvailabilityConditionAsExpression(availabilityToken: availabilityKeyword, negatedAvailabilityToken: negatedAvailabilityKeyword),
+        fixIts: [
+          FixIt(
+            message: ReplaceTokensFixIt(replaceTokens: [availabilityKeyword], replacement: negatedAvailabilityKeyword),
+            changes: [
+              .replace(oldNode: Syntax(availabilityKeyword), newNode: Syntax(negatedAvailabilityKeyword)),
+              .replaceTrailingTrivia(token: availability.rightParen, newTrivia: .zero),
+            ]
+          ),
+          FixIt(
+            message: RemoveNodesFixIt([comparisonOperator, falseKeyword]),
+            changes: .makeMissing([comparisonOperator, falseKeyword])
+          ),
```

Comment:
This should be a single `FixIt` with all the changes combined. As it stands right now we offer you two options to resolve the issue: Either to replace `#available` by `#unavailable` or by removing `== false`. But obviously, to resolve the error, you need to perform both changes.